test/unit/os/fs_spec.lua

```diff
@@ -432,55 +474,195 @@ describe('fs function', function()
     end)
 
     describe('os_open', function()
+      local new_file = 'test_new_file'
+      local existing_file = 'unit-test-directory/test_existing.file'
+
       before_each(function()
-        io.open('unit-test-directory/test_existing.file', 'w').close()
+        (io.open(existing_file, 'w')):close()
       end)
 
       after_each(function()
-        os.remove('unit-test-directory/test_existing.file')
-        os.remove('test_new_file')
+        os.remove(existing_file)
+        os.remove(new_file)
       end)
 
-      local new_file = 'test_new_file'
-      local existing_file = 'unit-test-directory/test_existing.file'
-
       it('returns UV_ENOENT for O_RDWR on a non-existing file', function()
         eq(ffi.C.UV_ENOENT, (os_open('non-existing-file', ffi.C.kO_RDWR, 0)))
       end)
 
-      it('returns non-negative for O_CREAT on a non-existing file', function()
+      it('returns non-negative for O_CREAT on a non-existing file which then can be closed', function()
         assert_file_does_not_exist(new_file)
-        assert.is_true(0 <= (os_open(new_file, ffi.C.kO_CREAT, 0)))
+        local fd = os_open(new_file, ffi.C.kO_CREAT, 0)
+        assert.is_true(0 <= fd)
+        eq(0, os_close(fd))
       end)
 
-      it('returns non-negative for O_CREAT on a existing file', function()
+      it('returns non-negative for O_CREAT on a existing file which then can be closed', function()
         assert_file_exists(existing_file)
-        assert.is_true(0 <= (os_open(existing_file, ffi.C.kO_CREAT, 0)))
+        local fd = os_open(existing_file, ffi.C.kO_CREAT, 0)
+        assert.is_true(0 <= fd)
```

Comment:
@justinmk `0 <= fd` is non-negative, not non-positive. Just for some reason it uses confusing order (constant first, I would write `fd >= 0`), I did not change this.